integration/kubernetes/helm-chart/alluxio/templates/proxy/statefulset.yaml

```diff
@@ -0,0 +1,117 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{- if .Values.proxy.enabled }}
+{{- $proxyCount := int .Values.proxy.count }}
+{{- $hostNetwork := .Values.proxy.hostNetwork }}
+{{- $hostPID := .Values.proxy.hostPID }}
+{{- $fullName := include "alluxio.fullname" . }}
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "alluxio.fullname" . }}-proxy
+  labels:
+    app: {{ template "alluxio.name" . }}
+    chart: {{ template "alluxio.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: alluxio-proxy
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "alluxio.name" . }}
+      chart: {{ template "alluxio.chart" . }}
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+      role: alluxio-proxy
+  serviceName: {{ $fullName }}-proxy
+  podManagementPolicy: Parallel
+  replicas: {{ $proxyCount }}
```

Comment:
if you want each master to have one proxy, you can use `master.count` instead of defining a separate `proxy.count`

Reply:
I don't think the Alluxio proxy is a stateful application afaik so I don't think we need to use StatefulSet.

Regarding the replicas, we ideally want them to be co-located with each worker (who are a `DaemonSet`). So I think you can just use `DaemonSet` and forget the `replicas` field.